src/components/Header/HeaderParts/NotificationButton.tsx

```diff
@@ -0,0 +1,320 @@
+'use client';
+
+import { useRef, useState, useEffect, MouseEvent, MutableRefObject } from 'react';
+import { useQuery, useQueryClient, useMutation } from '@tanstack/react-query';
+import { EventSourcePolyfill, Event } from 'event-source-polyfill';
+import { toast, ToastContainer } from 'react-toastify';
+import classNames from 'classnames/bind';
+
+import { useOutsideClick } from '@/hooks/useOutsideClick';
+import { NotificationIcon } from '@/public/index';
+import { getAlarm, postAlarmRead, deleteAlarm } from '@/api/alarmAPI';
+import { getCookie } from '@/libs/manageCookie';
+import type { AlarmAPIDataType, AlarmDataType, AlarmType } from '@/types/alarmTypes';
+import NotificationCard from './NotificationCard';
+
+import styles from './NotificationButton.module.scss';
+
+const cn = classNames.bind(styles);
+
+interface NotificationButtonProps {
+  isBlack: boolean;
+  eventSource: MutableRefObject<EventSource | null>;
+}
+
+const CATEGORY_LIST = ['전체', '상품', '이벤트', '커뮤니티'] as const;
+type CategoryType = (typeof CATEGORY_LIST)[number];
```

Comment:
왠지 enum형식을 사용하는거같은 느낌인데
```suggestion

enum CategoryType {
  All = '전체',
  Product = '상품',
  Event = '이벤트',
  Community = '커뮤니티',
}
```
지금 사용 방법도 나쁘지 않아보이는데 이런 방식이 더 깔끔할거같아요

Reply:
CATEGORY_LIST로 map을 돌리고 있어서, enum보다는 기존 코드가 더 좋지 않을까요...?

Reply:
단편적인 부분만 보고 말씀드린거라 사용해보시면서 판단하시면 될거같아요!
enum을 map으로 돌리는 경우는 해보지않아서 잘 모르겠습니다!